File: ABC/ABC376/A/main.go
package main

import "fmt"

func main() {
	var n, c int
	fmt.Scan(&n, &c)

	res := 0
	prev := -c
	for i := 0; i < n; i++ {
		var t int
		fmt.Scan(&t)
		if t >= prev+c {
			res++
			prev = t
		}
	}

	fmt.Println(res)
}

File: ABC/ABC376/A/main.cc
#include <iostream>

int main() {
  int n, c;
  std::cin >> n >> c;

  int res = 0, prev = -c;
  for (int i = 0; i < n; ++i) {
    int t;
    std::cin >> t;
    if (t >= prev + c) {
      ++res;
      prev = t;
    }
  }

  std::cout << res << "\n";

  return 0;
}